### PR TITLE
nextcloud19: init at 19.0.0

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

https://nextcloud.com/blog/nextcloud-hub-brings-productivity-to-home-office/

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by samueldr** [comment]
The wording of the warnings looks alright. Though I will not test as I am not using Nextcloud in the end.

**Comment by bachp** [comment]
@Ma27 When I to access a file after the upgrade a get an error:

```
\"Exception\":\"Sabre\\\\DAV\\\\Exception\\\\ServiceUnavailable\",\"Message\":\"filesystem not setup\"
```

**Comment by Ma27** [comment]
That's interesting. I experienced a similar behavior, but (wrongly) assumed that I screwed up my initial upgrade attempt.

In my case, Nextcloud reported that `The files of the app [all apps in /var/lib/nextcloud/apps] were not replaced correctly` and I also managed to fix this by restarting phpfpm.

Before merging this, we should at least find the cause for issue.

**Comment by ajs124** [comment]
We've also been observing weird issues with nextcloud where restarting phpfpm-nextcloud helps. For now, I've added the configured nextcloud package and some php stuff as a `restartTrigger` to it, as a workaround.

**Comment by bachp** [comment]
@ajs124 I was also thinking if adding `package` to the restart trigger would be the safest option.

**Comment by Ma27** [comment]
Sounds reasonable. I'd like to investigate the why before that though :)

**Comment by Ma27** [comment]
I managed to reproduce the issue in a minimal NixOS container using the configs from `nixos/tests/nextcloud/basic.nix`. Unfortunately I didn't find the culprit (neither in the code nor in the issue-tracker), but @ajs124's suggestion seems reasonable and fixed the issue in my local test environment.

I assume that most of the people hosting Nextcloud somewhere else are probably restarting these services anyways and didn't notice this.

**Comment by Ma27** [comment]
Rebased the patches onto latest master to resolve the merge-conflict in the release-notes.

**Comment by Ma27** [comment]
@ajs124 @bachp can you have another look? I'd merge this in a few days otherwise :)

**Review by bachp — APPROVED** [review]
Still works with the updated MR.

**Comment by Ma27** [comment]
Rebased onto master to resolve the merge-conflict. Anything else tbd or is this mergeable now?

File: nixos/modules/services/web-apps/nextcloud.nix
Comment (nlewo) [inline nixos/modules/services/web-apps/nextcloud.nix:373]: What about providing a bit of context?

```suggestion
        # When upgrading the Nextcloud package, Nextcloud can report errors such as 
        # "The files of the app [all apps in /var/lib/nextcloud/apps] were not replaced correctly"
        # Restarting phpfpm on Nextcloud package update fix these issues (but this is a workaround).
        phpfpm-nextcloud.restartTriggers = [ cfg.package ];
```

**Review by nlewo — APPROVED** [review]

**Comment by nlewo** [comment]
Thanks!

**Comment by Ma27** [comment]
Ported cdd1ecea80de3dc813fc86c6827af2c9a6bb50e7 to stable as documented in https://nixos.org/nixos/manual/index.html#module-services-nextcloud-maintainer-info

The module changes are targeted for 20.09 only.

**Pull request merged**
